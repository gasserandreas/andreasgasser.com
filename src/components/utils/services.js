/* global fetch */

function getUrl() {
  const url = process.env.GATSBY_BACKEND_URL || 'https://kx0kji7iyd.execute-api.us-east-1.amazonaws.com/prod/rest';
  return url;
}

export const requestInitialState = {
  data: null,
  pending: false,
  error: false,
  lastError: null,
  lastSuccess: null,
};

export async function sendContactMessage({ name, email, message }) {
  const url = getUrl();
  const config = {
    method: 'POST',
    mode: 'cors',
    cache: 'no-cache',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      name,
      email,
      message,
    }),
  };

  return fetch(url, config)
    .then((response) => response.json());
}

// eslint-disable-next-line
export const __testables__ = {
  getUrl,
};
