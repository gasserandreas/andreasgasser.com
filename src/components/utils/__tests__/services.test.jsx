/* global fetch */
import { __testables__, requestInitialState, sendContactMessage } from '../services';

const { getUrl } = __testables__;

describe('components/utils/services', () => {
  it('check requestInitialState', () => {
    expect(requestInitialState).toMatchSnapshot();
  });

  it('should return default url', () => {
    expect(getUrl).toMatchSnapshot();
  });

  it('should ', (done) => {
    fetch.mockResponseOnce('{"send":true,"data":"01000170f970fb94-0a203402-dfd0-427a-a023-c32ba13f29cf-000000"}');

    const person = {
      name: 'Test Name',
      email: 'test.name@domain.com',
      message: 'test message',
    };

    sendContactMessage(person)
      .then((data) => {
        expect(data).toMatchSnapshot();
        done();
      });
  });
});
