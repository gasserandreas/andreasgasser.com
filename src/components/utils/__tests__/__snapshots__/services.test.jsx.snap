// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/utils/services check requestInitialState 1`] = `
Object {
  "data": null,
  "error": false,
  "lastError": null,
  "lastSuccess": null,
  "pending": false,
}
`;

exports[`components/utils/services should  1`] = `
Object {
  "data": "01000170f970fb94-0a203402-dfd0-427a-a023-c32ba13f29cf-000000",
  "send": true,
}
`;

exports[`components/utils/services should return default url 1`] = `[Function]`;
